Source/DiabloUI/selgame.cpp

```diff
@@ -178,6 +182,10 @@ void selgame_GameSelection_Select(int value)
 	case 1: {
 		strncpy(title, _("Join TCP Games"), sizeof(title) - 1);
 
+#if defined __3DS__
+		ctr_vkbdInput(_("Enter address"), selgame_Ip, selgame_Ip);

```

Comment:
Is it not possible to move this to `UiInitList()` where the SDL, Switch and Vita implementation lives?

Reply:
So, the hint text was somewhat necessary because the keyboard applet blocks the current thread. Without that, it's difficult to determine what it is you're supposed to be entering via the keyboard. I ended up adding hint text to the `UiEdit` class and deferring 3DS keyboard input until after render and fade-in. I went ahead and applied the hint text to the Switch and Vita function calls as well.